### PR TITLE
feat: support backup configuration setting

### PR DESCRIPTION
**Pull request by poom**
## Pull Request Submission Checklist

Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

* support backup configuration setting

---
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-gcp-sql/12)

### CODE REVIEW

File: modules/google_sql_database_instance/variables.tf [hunk 1]
```diff
@@ -95,11 +95,35 @@ variable "settings_backup_configuration_enabled" {
...
+variable "settings_backup_configuration_transaction_log_retention_days" {
+  description = "(Optional) The number of days of transaction logs we retain for point in time restore, from 1-7."
+  type        = number
+  default     = 7
+}
+
+variable "settings_backup_configuration_start_time" {
+  description = "(Optional) HH:MM format time indicating when backup configuration starts."
+  type        = string
+  default     = "03:00"
```
Comment (deejiw) [inline modules/google_sql_database_instance/variables.tf]: where do we specify UTC for `settings_backup_configuration_start_time`

**Review by deejiw — APPROVED** [review]

File: modules/google_sql_database_instance/main.tf [hunk 2]
```diff
@@ -87,6 +99,8 @@ locals {
   instance_read_replica = var.name != "" ? "${var.name}-read-replica" : ""
 }
 
+#These setting will override from code
```
Comment (jai) [inline modules/google_sql_database_instance/main.tf]:
```suggestion
# This is a component module - these setting will be overridden from the embedding module/repo.
```

**Review by jai — CHANGES_REQUESTED** [review]
One small edit to the comment, thanks.

**Review by MXfive — COMMENTED** [review]
Just Jia's comment, then OK for me, ping again when fixed for a ✅

File: modules/google_sql_database_instance/variables.tf (hunk 1 above)
Comment (poom) [reply modules/google_sql_database_instance/variables.tf]: @deejiw @MXfive fixed

File: modules/google_sql_database_instance/main.tf (hunk 2 above)
Comment (poom) [reply modules/google_sql_database_instance/main.tf]: @jai I fixed the comment

**Review by MXfive — APPROVED** [review]

**Review by jai — APPROVED** [review]

**Pull request merged**